setup.py

```diff
@@ -174,15 +175,20 @@ def finalize_options(self):
 
     def run(self):
         import os
-        os.system("find . -name '*.pyc' | xargs rm -f")
-        os.system("rm -f python-build-stamp-2.4")
-        os.system("rm -f MANIFEST")
-        os.system("rm -rf build")
-        os.system("rm -rf dist")
-        os.system("rm -rf doc/_build")
-        os.system("rm -f sample.tex")
-
-
+        
+        for root, dirs, files in os.walk(os.path.dirname(os.path.realpath(__file__))):
+            for file in files:
+                if file.endswith('.pyc') and os.path.isfile:
+                    os.remove(os.path.join(root, file))
+
+        if os.path.isfile("python-build-stamp-2.4"): os.remove("python-build-stamp-2.4")
```

Comment:
Well, say you are in your home directory `~` and you have sympy in this directory `~/sympy/`. If I run `python ~/sympy/setup.py clean` from my home directory, it may delete every single `pyc` file in my home directory. Maybe disutils/setuptools prevent this, but it is always smart to be careful about `rm` commands, as you don't want to accidentally delete peoples files. It is worth checking that you can't delete files you don't intend to.

Also, how about a loop for this:

```python
filenames = [...]
for f in names:
    if os.path.isfile(f):
        os.remove(f)
    elif os.path.isdir(f):
        shutil.rmtree(f)
```